libs/event-manager/web-client/features/view-draft-event/src/lib/view-draft-event.component.ts

```diff
@@ -0,0 +1,709 @@
+          </form>
+          <div class="flex gap-2 w-full" style="margin-top: 1rem;">
+            <button
+              class="disco-btn red ease-in duration-300 text-lg uppercase border-4 px-8 py-2 cursor-pointer font-bold"
+              (click)="onCancel()"
+            >
+              Cancel
+            </button>
+            <button
+              class="disco-btn pink ease-in duration-300 text-lg uppercase border-4 px-8 py-2 cursor-pointer font-bold"
+              (click)="onSaveTickets()"
+            >
+              Save
+            </button>
+          </div>
+        </div>
+      </ng-container>
+    </div>
+  `,
+  styles: [
+    `
+      .advanced-pie-legend-wrapper {
+        display: none;
+      }
+    `,
+  ],
+  providers: [EventStore, ConfigStore],
+})
+export class ViewDraftEventComponent implements OnInit {
+  editingInfo = false;
+  editingDates = false;
+  editingTickets = false;
+  submitted = false;
+  startDate: number = 0;
+  endDate: number = 0;
+
+  readonly event$ = this._eventStore.event$;
+  readonly acceptedMintLogo$ = this._configStore.acceptedMintLogo$;
+  readonly acceptedMintSymbol$ = this._configStore.acceptedMintSymbol$;
+  readonly loading$ = this._eventStore.loading$;
+  readonly error$ = this._eventStore.error$;
+  readonly ticketsForm = this._formBuilder.group({
+    ticketQuantity: this._formBuilder.control(null, {
+      validators: [Validators.required],
+    }),
+    ticketPrice: this._formBuilder.control(null, {
+      validators: [Validators.required],
+    }),
+  });
+  readonly datesForm = this._formBuilder.group({
+    startDate: this._formBuilder.control(null, {
+      validators: [Validators.required],
+    }),
+    endDate: this._formBuilder.control(null, {
+      validators: [Validators.required],
+    }),
+  });
+
+  readonly infoForm = this._formBuilder.group({
+    name: this._formBuilder.control(null, {
+      validators: [Validators.required, Validators.maxLength(40)],
+    }),
+    description: this._formBuilder.control(null, {
+      validators: [Validators.required, Validators.maxLength(500)],
+    }),
+    location: this._formBuilder.control(null, {
+      validators: [Validators.required, Validators.maxLength(40)],
+    }),
+    banner: this._formBuilder.control(null, {
+      validators: [Validators.required, Validators.maxLength(40)],
+    }),
+  });
+
+  readonly now$ = interval(1000).pipe(
+    startWith(Date.now()),
+    map(() => Date.now())
+  );
+
+  readonly colorScheme = {
+    name: 'my-color-scheme',
+    selectable: false,
+    group: ScaleType.Linear,
+    domain: ['#19fb9b', '#ff5ef9'],
+  };
+
+  constructor(
+    private readonly _activatedRoute: ActivatedRoute,
+    private readonly _eventStore: EventStore,
+    private readonly _configStore: ConfigStore,
+    private readonly _formBuilder: UntypedFormBuilder,
+    private readonly _matSnackBar: MatSnackBar,
+    private readonly _router: Router
+  ) {
+    this.event$.subscribe((event) => {
```

Comment:
This is 100% a memory leak. [Make sure to unsubscribe](https://blog.bitsrc.io/6-ways-to-unsubscribe-from-observables-in-angular-ab912819a78f)

Why not create two new observable properties `startDate$` and `endDate$`, that transform the event into the proper data? And then using the async pipe you can forget about handling subscriptions.